BAU Added Temporary Header Forwarders for Rest API GW.

vam-gds opened the pull request:
<!-- Provide a general summary of your changes in the Title above -->
<!-- Include the Jira ticket number in square brackets as prefix, eg `[P4-XXXX] PR Title` -->

## Proposed changes

### What changed
BAU Added Temporary Header Forwarders for Rest API GW.

	modified:   deploy/template.yaml

<!-- Describe the changes in detail - the "what"-->

### Why did it change
BAU Added Temporary Header Forwarders for Rest API GW.

	modified:   deploy/template.yaml

<!-- Describe the reason these changes were made - the "why" -->

### Issue tracking
<!-- List any related Jira tickets or GitHub issues -->
<!-- List any related ADRs or RFCs -->
<!-- Delete/copy as appropriate -->

- [PYIC-XXXX](https://govukverify.atlassian.net/browse/PYIC-XXX)

## Checklists

### Environment variables or secrets

<!-- Delete if changes DO include new environment variables or secrets -->
- [ ] No environment variables or secrets were added or changed

<!-- Delete if changes DO NOT include new environment variables or secrets -->
- [ ] Documented in the [README](./blob/main/README.md)
- [ ] Added to deployment repository
- [ ] Added to local startup repository

### Other considerations

- [ ] Update [README](./blob/main/README.md) with any new instructions or tasks

vam-gds committed:
BAU Added Temporary Header Forwarders for Rest API GW.

	modified:   deploy/template.yaml

TobySaundersGDS committed:
Merge branch 'main' into BAU-Forwarding-HEADERS-RESTAPI

TobySaundersGDS committed:
Added x-forwarded-for header
As fallback option for adding client IP to session, REST API gateway adds x-forwarded-for header instead of 'forwarded' header

TobySaundersGDS committed:
Added x-forwarded-for header
As fallback option for adding client IP to session, REST API gateway adds x-forwarded-for header instead of 'forwarded' header

TalatAydinGDS commented on src/app/shared/ipAddressHelper.js:
```diff
@@ -10,6 +10,13 @@ module.exports = {
       } else {
         return unknown;
       }
+    } else if (req.headers && req.headers["x-forwarded-for"]) {
```
I think, the existing code can be simplified for better readability an maintainability.  I recommended below codes for example, 
module.exports = {
  getIpAddress: (req) => {
    const headerForward = req.headers && (req.headers["forwarded"] || req.headers["x-forwarded-for"]);
    if (headerForward) {
      const ipAddress = headerForward.match(IPV4_MATCH);
      if (ipAddress) {
        return ipAddress[0];
      }
    }
    return unknown;
  },
};

TalatAydinGDS reviewed (commented):
Overall, the change looks good. I just a made suggession.

TobySaundersGDS committed:
Added x-forwarded-for header
As fallback option for adding client IP to session, REST API gateway adds x-forwarded-for header instead of 'forwarded' header

TobySaundersGDS committed:
Added x-forwarded-for header
As fallback option for adding client IP to session, REST API gateway adds x-forwarded-for header instead of 'forwarded' header

TalatAydinGDS reviewed (approved):
LGTM
